apps/qubit/modules/informationobject/actions/uploadFindingAidAction.class.php

```diff
@@ -0,0 +1,66 @@
+<?php
+
+/*
+ * This file is part of the Access to Memory (AtoM) software.
+ *
+ * Access to Memory (AtoM) is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Access to Memory (AtoM) is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Access to Memory (AtoM).  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+class InformationObjectUploadFindingAidAction extends sfAction
+{
+  public function execute($request)
+  {
+    $this->resource = $this->getRoute()->resource;
+
+    // Check that object exists and that it is not the root
+    if (!isset($this->resource) || !isset($this->resource->parent))
+    {
+      $this->forward404();
+    }
+
+    // Check user authorization
+    if (!$this->context->user->isAuthenticated())
+    {
+      QubitAcl::forwardUnauthorized();
+    }
+
+    // Create form for file upload
+    $this->form = new sfForm;
+    $this->form->getValidatorSchema()->setOption('allow_extra_fields', true);
+    $this->form->setWidget('file', new sfWidgetFormInputFile);
```

Comment:
This could be a good scenario to use the `accept="application/pdf"` attribute? I think you can pass examples to the constructor or to `render()`.

- http://caniuse.com/#feat=input-file-accept
- http://www.wufoo.com/html5/attributes/07-accept.html